src/components/UserProfile/test/UserProfile.test.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import UserProfile from '..';
+
+describe('UserProfile', () => {
+  it('should match snapshot', () => {
+    const wrapper = shallow(<UserProfile profileData={{ firstname: 'Fistname' }} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+  it('should match snapshot', () => {
+    const wrapper = shallow(<UserProfile profileData={{ firstname: 'Fistname' }} user="owner" />);
+    expect(wrapper).toMatchSnapshot();
+  });
+});
+
+describe('UserProfile', () => {
+  const e = {
+      target: {
+        name: 'firstname',
+        value: 'Firstname',
+      },
+      preventDefault: () => {}
+    },
+    data = { name: 'gender', value: 'male' };
+  describe('handleChange', () => {
+    it('should set state property', () => {
+      const wrapper = shallow(<UserProfile profileData={{ firstname: 'Fistname' }} />);
+      wrapper.instance().handleChange(e);
+      expect(wrapper.state().firstname).toEqual('Firstname');
+    });
+  });
+
+  describe('handleSelect', () => {
+    it('should set state property', () => {
+      const wrapper = shallow(<UserProfile profileData={{ firstname: 'Fistname' }} />);
+      wrapper.instance().handleSelect(e, data);
+      expect(wrapper.state().gender).toEqual('male');
+    });
+  });
+
+  describe('handleSelect', () => {
+    it('should set state property', () => {
+      const wrapper = shallow(<UserProfile profileData={{ firstname: 'Fistname' }} />);
+      wrapper.instance().handleSelect(e, data);
+      expect(wrapper.state().gender).toEqual('male');
+    });
+  });
+
+  describe('handleSubmit', () => {
+    it('should call updateUser', () => {
+      const updateUserMock = jest.fn();
+      const wrapper = shallow(<UserProfile updateUser={updateUserMock} profileData={{ firstname: 'Fistname' }} />);
+      wrapper.instance().handleSubmit(e);
+      expect(updateUserMock.mock.calls.length).toBe(1);
```

Comment:
consider the suggestion below has its more descriptive of what is happening above.
```suggestion
      expect(updateUserMock).toHaveBeenCalledTimes(1);
```

https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber